core/src/main/java/org/opensearch/sql/expression/config/ExpressionConfig.java

```diff
@@ -51,8 +56,14 @@ public BuiltinFunctionRepository functionRepository() {
     return builtinFunctionRepository;
   }
 
+  @Bean
```

Comment:
FunctionProperties should be created per-request. Currently, Spring IoC container is created per-request which is not efficient and has issue (https://github.com/opensearch-project/sql/issues/611) , it will be changed to per jvm (https://github.com/opensearch-project/sql/pull/822), then BuiltinFunctionRepository is singleton instance per jvm.

Reply:
@penghuo in e7ccfc2bd2cc20c2c581210a58b5afda3ccf4d50 I moved FunctionProperties instantiation from ExpressionConfig to context creation time. Is that what you had in mind?

Reply:
Not exactally.
1. ApplicationContext should be created per-node instead of per-request. Which means FunctionProperties should not been created in ApplicationContext.
2. @dai-chen  also think of making FunctionRepositoy as  singleton.

I think FunctionProperties should be in a context and pass as paramater to Analyzer and FunctionProperties.

Reply:
@penghuo, I will move Spring setup to [`SQLPlugin.createComponents`](https://github.com/opensearch-project/sql/blob/bfad32cfe7ac3b8f2d8e8f889917a0e79be6c867/plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java#L123) and update as appropriate.

Reply:
> @penghuo, I will move Spring setup to [`SQLPlugin.createComponents`](https://github.com/opensearch-project/sql/blob/bfad32cfe7ac3b8f2d8e8f889917a0e79be6c867/plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java#L123) and update as appropriate.

As discussed offline, this PR will wait on `DSL` changes from `maximus-m1` branch.

Reply:
@penghuo and @dai-chen  this PR is ready for another review. The branch have been updated from `2.x` and modified to work with static `BuiltinFunctionRepository` class.

Thanks!